src/Grids/exterior_node.jl

```diff
@@ -0,0 +1,85 @@
+
+const c = Center()
+const f = Face()
+
+const XBoundedGrid = AbstractGrid{<:Any, <:Bounded}
+const YBoundedGrid = AbstractGrid{<:Any, <:Any, <:Bounded}
+const ZBoundedGrid = AbstractGrid{<:Any, <:Any, <:Any, <:Bounded}
+
+const XYBoundedGrid = AbstractGrid{<:Any, <:Bounded, <:Bounded}
+const XZBoundedGrid = AbstractGrid{<:Any, <:Bounded, <:Any, <:Bounded}
+const YZBoundedGrid = AbstractGrid{<:Any, <:Any, <:Bounded, <:Bounded}
+
+const XYZBoundedGrid =  AbstractGrid{<:Any, <:Bounded, <:Bounded, <:Bounded}
+
+#####
+##### Exterior node and peripheral node
+#####
+
+"""
+    exterior_cell(i, j, k, grid)
+
+Return `true` when the tracer cell is outside the boundaries of the domain.
+Return `false` either when the tracer cell at index `i, j, k` is "wet".
+In non-`Bounded` directions, `exterior_node` always returns `false`.
+"""
+@inline exterior_cell(i, j, k, grid) = false
+@inline exterior_cell(i, j, k, grid::XBoundedGrid) = (i < 1) | (i > grid.Nx)
+@inline exterior_cell(i, j, k, grid::YBoundedGrid) = (j < 1) | (j > grid.Ny)
+@inline exterior_cell(i, j, k, grid::ZBoundedGrid) = (k < 1) | (k > grid.Nz)
+
+@inline exterior_cell(i, j, k, grid::XYBoundedGrid) = (i < 1) | (i > grid.Nx) | (j < 1) | (j > grid.Ny)
+@inline exterior_cell(i, j, k, grid::XZBoundedGrid) = (i < 1) | (i > grid.Nx) | (k < 1) | (k > grid.Nz)
+@inline exterior_cell(i, j, k, grid::YZBoundedGrid) = (j < 1) | (j > grid.Ny) | (k < 1) | (k > grid.Nz)
+
+@inline exterior_cell(i, j, k, grid::XYZBoundedGrid) = (i < 1) | (i > grid.Nx) |
+                                                       (j < 1) | (j > grid.Ny) |
+                                                       (k < 1) | (k > grid.Nz)
+
+"""
+    exterior_node(LX, LY, LZ, i, j, k, grid)
+
+Return `true` when the location `(LX, LY, LZ)`, interpreted as _either_ a
+cell interface, or the cell itself, is "entirely" exterior to a `Bounded` domain.
```

Comment:
Same comment here as above. Does this include include immersed solid?

Reply:
This function defines `exterior_node` for "native" or "underlying" grids. Later, in `ImmersedBoundaries`, this function is extended to also be correct when an `ImmersedBoundaryGrid` is wrapped around an underlying grid.